android/src/main/java/io/branch/rnbranch/RNBranchModule.java

```diff
@@ -60,22 +71,57 @@ public void onInitFinished(JSONObject referringParams, BranchError error) {
                 Log.d(REACT_CLASS, "onInitFinished");
                 JSONObject result = new JSONObject();
                 try{
-                    result.put("params", referringParams != null && referringParams.has("~id") ? referringParams : JSONObject.NULL);
-                    result.put("error", error != null ? error.getMessage() : JSONObject.NULL);
-                    result.put("uri", uri != null ? uri.toString() : JSONObject.NULL);
+                    result.put(NATIVE_INIT_SESSION_FINISHED_EVENT_PARAMS, referringParams != null && referringParams.has("~id") ? referringParams : JSONObject.NULL);
+                    result.put(NATIVE_INIT_SESSION_FINISHED_EVENT_ERROR, error != null ? error.getMessage() : JSONObject.NULL);
+                    result.put(NATIVE_INIT_SESSION_FINISHED_EVENT_URI, uri != null ? uri.toString() : JSONObject.NULL);
                 } catch(JSONException ex) {
                     try {
                         result.put("error", "Failed to convert result to JSONObject: " + ex.getMessage());
                     } catch(JSONException k) {}
                 }
                 initSessionResult = result;
-                LocalBroadcastManager.getInstance(mmActivity).sendBroadcast(new Intent(NATIVE_INIT_SESSION_FINISHED_EVENT));
+
+                BranchUniversalObject branchUniversalObject =  BranchUniversalObject.getReferredBranchUniversalObject();
```

Comment:
Can we use`Branch.BranchUniversalReferralInitListener()` instead of `Branch.BranchReferralInitListener()`. This callback is already giving `BranchUniversalObject` and `LinkProperties`
```branch.initSession(new Branch.BranchUniversalReferralInitListener() {
            @Override
            public void onInitFinished(BranchUniversalObject branchUniversalObject, LinkProperties linkProperties, BranchError error) ```